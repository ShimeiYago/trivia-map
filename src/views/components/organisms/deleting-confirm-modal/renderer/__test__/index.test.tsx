import { shallow, ShallowWrapper } from 'enzyme';
import { Renderer, Props } from '..';

let shallowWrapper: ShallowWrapper<Props, unknown, Renderer>;

const props: Props = {
  open: true,
  markerDeletingState: 'waiting',
  onClickCancel: jest.fn(),
  onClickConfirm: jest.fn(),
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    shallowWrapper = shallow(<Renderer {...props} />);
  });

  it('basic', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('loading', () => {
    shallowWrapper.setProps({
      open: false,
      markerDeletingState: 'loading',
    });
    expect(shallowWrapper).toMatchSnapshot();
  });
});
