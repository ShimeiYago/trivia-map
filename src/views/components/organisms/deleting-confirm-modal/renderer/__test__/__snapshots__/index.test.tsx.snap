// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<ForwardRef(Dialog)
  open={true}
>
  <ForwardRef(DialogTitle)>
    この投稿を削除してもよろしいですか？
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogActions)>
    <ForwardRef(Button)
      onClick={[MockFunction]}
      variant="outlined"
    >
      キャンセル
    </ForwardRef(Button)>
    <ForwardRef(LoadingButton)
      autoFocus={true}
      loading={false}
      onClick={[MockFunction]}
      variant="contained"
    >
      OK
    </ForwardRef(LoadingButton)>
  </ForwardRef(DialogActions)>
</ForwardRef(Dialog)>
`;

exports[`Shallow Snapshot Tests loading 1`] = `
<ForwardRef(Dialog)
  open={true}
>
  <ForwardRef(DialogTitle)>
    この投稿を削除してもよろしいですか？
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogActions)>
    <ForwardRef(Button)
      onClick={[MockFunction]}
      variant="outlined"
    >
      キャンセル
    </ForwardRef(Button)>
    <ForwardRef(LoadingButton)
      autoFocus={true}
      loading={true}
      onClick={[MockFunction]}
      variant="contained"
    >
      OK
    </ForwardRef(LoadingButton)>
  </ForwardRef(DialogActions)>
</ForwardRef(Dialog)>
`;
